star-rx72n-firmware/lib/rx_frame/src/rx_frame.c

```diff
@@ -0,0 +1,284 @@
+/**
+ * @file rx_frame.c
+ * @brief Frame Layer Implementation
+ *
+ * Implements frame encoding/decoding with CRC-32 verification.
+ * Bit-exact compatible with star-gateway/internal/frame/.
+ *
+ * STAR Project - Texas A&M University
+ * December 2025
+ */
+
+#include "rx_frame.h"
+
+#include <string.h>
+
+/* =============================================================================
+ * Private Helper Functions
+ * =============================================================================
+ */
+
+/**
+ * @brief Write uint16 in big-endian format
+ *
+ * @param[out] buf Output buffer (at least 2 bytes)
+ * @param[in]  val Value to write
+ */
+static void internal_write_be16(uint8_t *buf, uint16_t val)
+{
+    buf[0] = (uint8_t)(val >> 8);
+    buf[1] = (uint8_t)(val & 0xFF);
+}
+
+/**
+ * @brief Read uint16 in big-endian format
+ *
+ * @param[in] buf Input buffer (at least 2 bytes)
+ * @return Decoded value
+ */
+static uint16_t internal_read_be16(const uint8_t *buf)
+{
+    return ((uint16_t)buf[0] << 8) | (uint16_t)buf[1];
+}
+
+/**
+ * @brief Write uint32 in little-endian format (for CRC-32)
+ *
+ * @param[out] buf Output buffer (at least 4 bytes)
+ * @param[in]  val Value to write
+ */
+static void internal_write_le32(uint8_t *buf, uint32_t val)
+{
+    buf[0] = (uint8_t)(val & 0xFF);
+    buf[1] = (uint8_t)((val >> 8) & 0xFF);
+    buf[2] = (uint8_t)((val >> 16) & 0xFF);
+    buf[3] = (uint8_t)((val >> 24) & 0xFF);
+}
```

Comment:
we shouldnt be indexing with numbers, this should be an enum so we understand what each index represents